### PR TITLE
FLPATH-405: etag & if-not-modified

### PR DESCRIPTION
**Pull request by RichardW98**
**What this PR does / why we need it**:
return 304 if the value of  if-not-modified in request header is equal to etag
**Which issue(s) this PR fixes**:
Fixes #405

**Change type**
- [ ] New feature
- [ ] Bug fix
- [ ] Unit tests
- [ ] Integration tests
- [ ] CI
- [ ] Documentation
- [ ] Auto-generated SDK code

**Impacted services**
- [ ] Workflow Service
- [ ] Notification Service

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.

### CODE REVIEW

File: workflow-service/src/main/java/com/redhat/parodos/workflow/execution/dto/WorkFlowStatusResponseDTO.java [hunk 1]
```diff
@@ -45,6 +46,6 @@ public class WorkFlowStatusResponseDTO {
 	private WorkStatus status;
 
 	@JsonInclude(JsonInclude.Include.NON_EMPTY)
-	private List<WorkStatusResponseDTO> works;
+	private transient List<WorkStatusResponseDTO> works;
```
Comment (masayag) [inline workflow-service/src/main/java/com/redhat/parodos/workflow/execution/dto/WorkFlowStatusResponseDTO.java:49]: if this one is `transient`, shouldn't the annotation be changed to `@JsonIgnore` or should it remain serialized to json?

Comment (RichardW98) [reply workflow-service/src/main/java/com/redhat/parodos/workflow/execution/dto/WorkFlowStatusResponseDTO.java:49]: we need it to be serialized.  sonarlint requires jsonignore or transient, so I have to pick one

Comment (RichardW98) [reply workflow-service/src/main/java/com/redhat/parodos/workflow/execution/dto/WorkFlowStatusResponseDTO.java:49]: this is dto, so I think transient won't afffect

File: workflow-service/src/test/java/com/redhat/parodos/project/controller/ProjectControllerTest.java [hunk 2]
```diff
@@ -206,7 +220,7 @@ public void testRemoveUsersFromProject_WithEmptyUsersList() throws Exception {
 
 	ProjectResponseDTO createSampleProject(String name) {
 		ProjectResponseDTO responseDTO = new ProjectResponseDTO();
-		responseDTO.setId(UUID.randomUUID());
+		responseDTO.setId(UUID.fromString("af7c1fe6-d669-414e-b066-e9733f0de7a8"));
```
Comment (gciavarrini) [inline workflow-service/src/test/java/com/redhat/parodos/project/controller/ProjectControllerTest.java:223]: Some tests use this method.
WDYT to set the "hard coded" `Id` in  `testListProjects_when_etagEquals_then_return_notModified` test instead?

Comment (RichardW98) [reply workflow-service/src/test/java/com/redhat/parodos/project/controller/ProjectControllerTest.java:223]: I was expecting the hash of the same UUID string should always return the same value. but different java version may have different result, I will rollback to randomUUID

Comment (gciavarrini) [reply workflow-service/src/test/java/com/redhat/parodos/project/controller/ProjectControllerTest.java]: If we want to track this, maybe we can add another unit tests. So we can be sure that each generated id will not change

**Review by masayag — APPROVED** [review]

File: workflow-service/src/main/java/com/redhat/parodos/workflow/execution/dto/WorkFlowStatusResponseDTO.java (hunk 1 above)
Comment (RichardW98) [reply workflow-service/src/main/java/com/redhat/parodos/workflow/execution/dto/WorkFlowStatusResponseDTO.java]: i rolled back all the serialzable implement.   obj.hashcode() doesn't need it

**Pull request merged**